documentation/blog/2025-09-26-subagents-vs-subrecipes/index.md

```diff
@@ -0,0 +1,130 @@
+
+* **Setup complexity** -- Requires careful YAML file creation and parameter definition.
+* **Learning curve** -- Need to understand recipe syntax and structure.
+* **File management** -- Must organize and maintain recipe files.
+
+
+## A Decision Framework
+
+**Use subagents when:**
+- You need quick, one-off task delegation
+- Tasks can be completed independently and don't need to be repeated
+
+**Use subrecipes when:**
+- Building reusable workflows you can share with your team
+- Need structured parameter handling
+
+## Common Benefits and Limitations of Both Approaches
+
+Both of these features share some limitations. We've already mentioned the experimental and evolving nature, but there are a few more important ones to note:
+
+* Subrecipes and subagents run their tasks in isolation and do not share state with one another. This isolation helps prevent conflicts and keeps tasks self-contained. If you do need to share information between the processes, you will have to be very explicit about this in your prompts and instructions.
+
+* Neither can spawn additional similar workers: subagents can't create more subagents, and subrecipes can't call other subrecipes. This prevents runaway processes but limits deep nesting.
+
+* Whether you're using subagents or subrecipes, you can only run up to 10 concurrent parallel workers total. This is not user-configurable, but this limit keeps resource usage manageable; this may limit very large-scale parallelism.
+
+
+## Getting Started
+
+My recommendation: start with subagents to experiment and understand your workflow needs. They're easier to get started with because you don't need to write configuration files first.
+
+Once you identify patterns you want to repeat, you can convert that subagent workflow session into a recipe and subrecipe structure. This gives you the best progression from experimentation to production.
+
+## The Choice is Yours
+
+The choice depends on your specific needs and workflow requirements. Quick tasks that won't be repeated favor subagents. Complex workflows with multiple steps or customization favor subrecipes.
+
+Your workflow requirements should drive the decision.
+
+Share your subagent prompts or subrecipe ideas with us on our [Discord community](https://discord.gg/block-opensource) or [GitHub discussions](https://github.com/block/goose/discussions).
+
+
+
+<head>
+  <meta property="og:title" content="How to Choose Between Subagents and Subrecipes in goose" />
+  <meta property="og:type" content="article" />
+  <meta property="og:url" content="https://block.github.io/goose/blog/2025-09-26-subagents-vs-subrecipes" />
+  <meta property="og:description" content="When you need to break complex work into multiple AI tasks, should you use subagents or subrecipes? Learn the key differences and when to use each approach." />
+  <meta property="og:image" content="https://block.github.io/goose/assets/images/subagents-vs-subrecipes-19bca16b86a951e4618be8ab6ce90fb2.png" />
```

Comment:
this image doesnt end with an s, but the one below twitter:image does..double check the url to figure out which one it should . I'm referring to this part

`9bca16b86a951e4618be8ab6ce90fb2.png` vs `9bca16b86a951e4618be8ab6ce90fb2s.png`